cli/README.md

```diff
@@ -0,0 +1,189 @@
+# Cli - Mamba
+
+The CLI pkg also named Mamba.
+
+## About the project
+
+Mamba provide a higher level and more friendly interfaces to build modern command line interfaces and manage configurations for Go applications.
+
+It bases on
+
+-   [spf13/pflag](https://github.com/spf13/pflag): to provide POSIX/GNU-style flags
+-   [spf13/cobra](https://github.com/spf13/cobra): to provide a simple interface to create powerful modern CLI interfaces
+-   [spf13/viper](https://github.com/spf13/viper): to provide a complete configuration sulution for [12-Factor](https://12factor.net/zh_cn/) apps
+
+## Status
+
+**Working in process**
+
+## Concepts
+
+A cli is composed of commands, argumetns & flags.
+
+**Commands** represent actions, **Args** are things and **Flags** are modifiers for those actions.
+
+The pattern to follow is `appname commands args --flag`
+
+The flag is compatible with the [GNU extensions to the POSIX recommendations for command-line options](http://www.gnu.org/software/libc/manual/html_node/Argument-Syntax.html).
```

Comment:
the flags follow the style mentioned in... ?

Reply:
copy from pflag